asynction/exceptions.py

```diff
@@ -38,3 +38,28 @@ class MessageAckValidationException(ValidationException):
     """
 
     pass
+
+
+class SecurityException(AsynctionException):
+    """
+    Base Security Exception type.
+    """
+
+    pass
+
+
+class UnregisteredSecurityScheme(SecurityException):
+    """
+    Raised when a security scheme not listed in the securitySchemes section of the
+    spec is used in a ``security`` or ``x-security`` specification
+    """
+
+    pass
+
+
+class UnsupportedSecurityScheme(SecurityException):
+    """
+    Raised when a specified security scheme is not supported by asynction
+    """
+
+    pass
```

Comment:
The `exceptions.py` module is treated as part of the Asynction user facing API and is meant to include the exceptions that an event handler may raise, because it is wrapped within a schema or security validator.

These exceptions are well known and defined so that the user can act upon them as part of their Flask-socketIO error handlers. See https://flask-socketio.readthedocs.io/en/latest/getting_started.html#error-handling

The `UnregisteredSecurityScheme` is essentially a deserialisation exception, rather than an event handler exception, so it shouldn't belong to this module. IMHO we should just use a ValueError, which is the exc type we already use for any other deserialisation error. See my comments under the `AsyncAPISpec` class of `types.py`.

The `UnsupportedSecurityScheme` can also be removed, since we do not need to fail hard on such scenarios. See my comments under `security.py`.